config/v1/types_image.go

```diff
@@ -83,6 +108,14 @@ type ImageStatus struct {
 	// field in ImageStreams. The value must be in "hostname[:port]" format.
 	// +optional
 	ExternalRegistryHostnames []string `json:"externalRegistryHostnames,omitempty"`
+
+	// imageStreamImportMode controls the import mode behaviour of imagestreams. It can be
+	// `Legacy` or `PreserveOriginal` depending on the payload type of the cluster or the
+	// `ImageStreamImportMode` setting in the spec. This value is set by the image registry
+	// operator.
+	// +openshift:enable:FeatureGate=ImageStreamImportMode
+	// +optional
+	ImageStreamImportMode ImportModeType `json:"imageStreamImportMode,omitempty"`
```

Comment:
Can you explain the relationship between the spec and the status version of these fields?

Reply:
the status will be reconciled based on:
1. the spec value 
2. if no spec value is specified, the image registry operator would look at the ClusterVersion status to determine the payload arch (TBD) and set the import mode accordingly, i.e single arch payload => the import mode is Legacy and multi payload => PreserveOriginal.

Reply:
I think those details should make it into the godoc for the field

Reply:
added.